qiskit_ionq/ionq_job.py

```diff
@@ -255,9 +256,10 @@ def result(self):
         Returns:
             Result: A Qiskit :class:`Result <qiskit.result.Result>` representation of this job.
         """
-        # Short-circuit if we have already cached the result for this job.
-        if self._result is not None:
-            return self._result
+        # TODO: cache results by aggregation type
+
+        if aggregation is not None and not isinstance(aggregation, AggregationType):
```

Comment:
Can we warn instead of throw here?

Reply:
why not throw ? the request would fail otherwise.

Reply:
Relying on server-side validation means we won't need to update the client when we add a new aggregation type.

Same issue with doing target totally client side, you either need to wire up to an endpoint that lets you know the available options (for targets, /backends), or you're heavily restricting what the client can do for something that should be checked server-side anyway.

Reply:
Got it. Makes sense.